### DISCUSSION TITLE
hide commits from bot in changelog

### DISCUSSION THREAD

**Discussion by swaroopar** [jreleaser/jreleaser#1542] [Q&A] [answered]
Hi All,

We wanted to hide commits from a specific bot user in the change log and added the necessary config - https://github.com/eclipse-xpanse/terraform-boot/blob/main/jreleaser.yml#L23

But the changelog does not hide these commits. - https://github.com/eclipse-xpanse/terraform-boot/releases/tag/v1.0.3

Job output attached. Could you please let me know what is causing this?

[jreleaser-release (1).zip](https://github.com/jreleaser/jreleaser/files/13512237/jreleaser-release.1.zip)

**Comment by aalmiray** [comment]
AFAICT contributor names are matched using this condition https://github.com/jreleaser/jreleaser/blob/e9199ff6c4e084aaca5ab4ce550e3852190cbce6/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/release/Changelog.java#L1192

You could try switching to a regex. I'd recommend running the `changelog` command in a local environment to see if the changes work as expected, before pushing to live.

**Comment by swaroopar** [comment]
@aalmiray thank you for looking into it. Even adding the regular expression did not help.  i used the expression `^eclipse-xpanse-bot.*$`

![image](https://github.com/jreleaser/jreleaser/assets/54129659/a4d9d63e-0b00-42ad-991e-b15da7349294)

Also, looking at this example, it should work even without regex - https://github.com/jreleaser/jreleaser/blob/main/jreleaser.yml#L106

**Reply by aalmiray** [reply, marked as answer]
Hmm hold on a second. Re-reading the title what is expected is to hide commits, not contributor names which is what the configuration you have in place will do. 

First I thought we could use a labeler & category to hide target commits by contributor/author/committer but that’s not possible at the moment as labelers are checked against commit title & message. 

Hiding commits by contributor is a missing feature.

**Reply by swaroopar** [reply]
@aalmiray oh ok. Thanks for clarifying, I misunderstood the documentation then. 

Would it be possible to achieve this by labeling commits based on title/body and then use the excludeLabels option to filter it out from changelog?

**Reply by swaroopar** [reply]
Yes, this option works after adding `formatted: ALWAYS`
Thanks again @aalmiray

**Reply by aalmiray** [reply]
Yes, you'll have to use a labeler matching title and/or body; assign such label to a category; hide the chosen category.

I'm thinking what config options may be available in the future to enable the following:

- hide contributor name but show commits (what it's possible today)
- hide contributor name and matching commits

I think it's not worth hiding commits but show contributor name.

**Reply by aalmiray** [reply]
Filed and fixed https://github.com/jreleaser/jreleaser/issues/1543

**Reply by swaroopar** [reply]
@aalmiray Great thank you. I will switch to the new option as soon as it is available.
